pos_require_product_quantity/static/src/js/Screens/ProductScreen/ProductScreen.esm.js

```diff
@@ -0,0 +1,41 @@
+/** @odoo-module **/
+
+/*
+  Copyright 2019 Coop IT Easy SCRLfs
+    Robin Keunen <robin@coopiteasy.be>
+    Simon Hick <simon.hick@coopiteasy.be>
+  License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+*/
+
+import ProductScreen from "point_of_sale.ProductScreen";
+import Registries from "point_of_sale.Registries";
+
+const PosRequireProductQuantityProductScreen = (OriginalProductScreen) =>
+    class extends OriginalProductScreen {
+        async _onClickPay() {
+            const result = super._onClickPay(...arguments);
```

Comment:
i think that this should be called after the rest of the checks, because it will try to switch to the payment screen. if a check fails, it is better to not run this code at all.